src/dbnode/storage/flush.go

```diff
@@ -200,12 +169,63 @@ func (m *flushManager) Flush(
 		}
 		multiErr = multiErr.Add(ns.FlushIndex(indexFlush))
 	}
-	// mark index flush finished
 	multiErr = multiErr.Add(indexFlush.DoneIndex())
 
 	return multiErr.FinalError()
 }
 
+func (m *flushManager) rotateCommitlogAndSnapshot(
+	namespaces []databaseNamespace,
+	tickStart time.Time,
+) error {
+	rotatedCommitlogID, err := m.commitlog.RotateLogs()
+	if err != nil {
+		return fmt.Errorf("error rotating commitlog in mediator tick: %v", err)
+	}
+
+	snapshotID := uuid.NewUUID()
+
+	snapshotPersist, err := m.pm.StartSnapshotPersist(snapshotID)
+	if err != nil {
+		return err
+	}
+
+	m.setState(flushManagerSnapshotInProgress)
+	var (
+		maxBlocksSnapshottedByNamespace = 0
+		multiErr                        = xerrors.NewMultiError()
+	)
+	for _, ns := range namespaces {
+		snapshotBlockStarts := m.namespaceSnapshotTimes(ns, tickStart)
+
+		if len(snapshotBlockStarts) > maxBlocksSnapshottedByNamespace {
+			maxBlocksSnapshottedByNamespace = len(snapshotBlockStarts)
+		}
+		for _, snapshotBlockStart := range snapshotBlockStarts {
+			err := ns.Snapshot(
+				snapshotBlockStart, tickStart, snapshotPersist)
+
+			if err != nil {
+				detailedErr := fmt.Errorf("namespace %s failed to snapshot data: %v",
+					ns.ID().String(), err)
+				multiErr = multiErr.Add(detailedErr)
+			}
+		}
+	}
+	m.maxBlocksSnapshottedByNamespace.Update(float64(maxBlocksSnapshottedByNamespace))
+
+	err = snapshotPersist.DoneSnapshot(snapshotID, rotatedCommitlogID)
+	if err != nil {
+		multiErr = multiErr.Add(err)
+	}
```

Comment:
nit: instead of `err != nil { multiErr = multiErr.Add(err) }` instead just:
```go
multiErr = multiErr.Add(snapshotPersist.DoneSnapshot(snapshotID, rotatedCommitlogID))
```

Since it will do the err == nil check for you?

Reply:
yeah @justinjc had the same comment. I prefer the explicit error handling but I'll change it